tree/ntuple/v7/src/RField.cxx

```diff
@@ -916,6 +923,181 @@ void ROOT::Experimental::RClassField::AcceptVisitor(Detail::RFieldVisitor &visit
 
 //------------------------------------------------------------------------------
 
+ROOT::Experimental::RCollectionClassField::RCollectionClassField(std::string_view fieldName, std::string_view className)
+   : RCollectionClassField(fieldName, className, TClass::GetClass(std::string(className).c_str()))
+{
+}
+
+ROOT::Experimental::RCollectionClassField::RCollectionClassField(std::string_view fieldName, std::string_view className,
+                                                                 TClass *classp)
+   : ROOT::Experimental::Detail::RFieldBase(fieldName, className, ENTupleStructure::kCollection, false /* isSimple */),
+     fNWritten(0)
+{
+   if (classp == nullptr)
+      throw RException(R__FAIL("RField: no I/O support for collection proxy type " + std::string(className)));
+
+   fProxy = classp->GetCollectionProxy();
+   if (!fProxy)
+      throw RException(R__FAIL(std::string(className) + " has no associated collection proxy"));
+   if (fProxy->HasPointers())
+      throw RException(R__FAIL("collection proxies whose value type is a pointer are not supported"));
+   if (fProxy->GetProperties() & TVirtualCollectionProxy::kIsAssociative)
+      throw RException(R__FAIL("associative collections not supported"));
+
+   std::unique_ptr<ROOT::Experimental::Detail::RFieldBase> itemField;
+   if (auto valueClass = fProxy->GetValueClass()) {
+      // Element type is a class
+      itemField = RFieldBase::Create("_0", valueClass->GetName()).Unwrap();
+   } else {
+      switch (fProxy->GetType()) {
+      case EDataType::kChar_t:   itemField = std::make_unique<RField<char>>("_0"); break;
+      case EDataType::kUChar_t:  itemField = std::make_unique<RField<std::uint8_t>>("_0"); break;
+      case EDataType::kShort_t:  itemField = std::make_unique<RField<std::int16_t>>("_0"); break;
+      case EDataType::kUShort_t: itemField = std::make_unique<RField<std::uint16_t>>("_0"); break;
+      case EDataType::kInt_t:    itemField = std::make_unique<RField<std::int32_t>>("_0"); break;
+      case EDataType::kUInt_t:   itemField = std::make_unique<RField<std::uint32_t>>("_0"); break;
+      case EDataType::kLong_t:
```

Comment:
On 32 bits platforms, long is only 32 bits, is that supported by this?

Reply:
I tried to match the type translation table used in `RFieldBase::Create()` (see [here](https://github.com/root-project/root/blob/master/tree/ntuple/v7/src/RField.cxx#L77)), but I agree this might be problematic.

Reply:
@jblomer Is this an issue?

Reply:
Yes, I think that is an issue, thanks for pointing it out! @jalopezg-r00t could you create an issue on GH?

Reply:
This is now https://github.com/root-project/root/issues/11616.